opensearch_single_kernel/events/opensearch.py

```diff
@@ -0,0 +1,675 @@
+                return
+        # apply the directives computed and emitted by the peer cluster manager
+        if not self._apply_peer_cm_directives_and_check_if_can_start():
+            self.logger.debug("cannot start peer cm had a blocking directive")
+            event.defer()
+            return
+
+        if (
+            not self.charm.state.application.is_admin_user_configured
+            or not self.charm.tls_manager.is_fully_configured()
+        ):
+            if not self.charm.state.tls_relation:
+                status = CharmStatuses.TLS_RELATION_MISSING.value
+            else:
+                if self.charm.state.application.is_admin_user_configured:
+                    status = CharmStatuses.ADMIN_USER_NOT_CONFIGURED.value
+                else:
+                    status = CharmStatuses.TLS_NOT_FULLY_CONFIGURED.value
+            self.charm.status.set(status)
+            event.defer()
+            return
+
+        self.charm.status.clear(CharmStatuses.ADMIN_USER_NOT_CONFIGURED.value)
+        self.charm.status.clear(CharmStatuses.TLS_NOT_FULLY_CONFIGURED.value)
+        self.charm.status.clear(CharmStatuses.TLS_RELATION_MISSING.value)
+
+        if self.charm.state.unit.is_app_leader:
+            self.charm.status.clear(CharmStatuses.PEER_CLUSTER_NO_RELATION, app=True)
+
+        # Configure OpenSearch Users
+        if not self.charm.state.server.is_app_leader:
+            self.charm.users_manager.purge_initial_users()
+            for user in OPENSEARCH_SYSTEM_USERS:
+                user_hash = self.charm.state.secrets.hash_key(user)
+                hashed_pwd = self.charm.state.secrets.get(Scope.APP, user_hash)
+                self.charm.users_manager.save_user_locally(user, hashed_pwd)
+
+        # Configure Client Authentication
+        self.charm.cluster_manager.set_client_auth()
+
+        deployment_desc = self.charm.state.application.deployment_desc
+        # only start the main orchestrator if a data node is available
+        # this allows for "cluster-manager-only" nodes in large deployments
+        # workflow documentation:
+        # no "data" role in deployment desc -> start gets deferred
+        # when "data" node joins -> start cluster-manager via _on_peer_cluster_relation_changed
+        # cluster-manager notifies "data" node via refresh of peer cluster relation data
+        # "data" node starts and initializes security index
+        if (
+            deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            and not deployment_desc.start == StartMode.WITH_GENERATED_ROLES
+            and "data" not in deployment_desc.config.roles
+            and not self.charm.state.application.security_index_initialised
+        ):
+            self.charm.status.set(CharmStatuses.PEER_CLUSTER_NO_DATA_NODE.value)
+            event.defer()
+            return
+        # We are requesting start of openSearch
+        self.charm.status = CharmStatuses.REQUEST_LOCK_ON_START.value
```

Comment:
```suggestion
         self.charm.status.set(CharmStatuses.REQUEST_LOCK_ON_START.value)
```
I believe, status setting should be done using set method.